### PR TITLE
Upgrade to GraalVM 20.3.1

### PR DESCRIPTION
**Pull request by zakkak**
Replaces #14455 

Copying @gsmet's original PR body:
> Want to upgrade to 20.3.1 first as we will probably backport it to 1.11.
>
> Creating as draft for now to see how it goes in my fork.

Reverts changes made to work around issues introduced by 20.3.0 that are no longer present in 20.3.1 (see https://github.com/quarkusio/quarkus/pull/13644).
More specifically it reverts:
- https://github.com/quarkusio/quarkus/pull/12912 (9e0a94ebf07021a1d95050a369c8af340d153dcf)
- https://github.com/quarkusio/quarkus/pull/13409 (af31023b4a76ce6e5b14e3db97b6b679febf0ac8)
- https://github.com/quarkusio/quarkus/pull/13644 (aa7b253b93857ef6887c6e0e5f2701cce0d6def9, 462eff2daa3c09fa6cf5bd514b34a486fcf947fb, 1fb47db670655e44ad174c907f5ee6cf53b7160c)

### CODE REVIEW

File: extensions/tika/runtime/src/main/java/io/quarkus/tika/graalvm/CleanerNotSupportedSubstitution.java [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package io.quarkus.tika.runtime.graal;
+package io.quarkus.tika.graalvm;
```
Comment (gsmet) [inline extensions/tika/runtime/src/main/java/io/quarkus/tika/graalvm/CleanerNotSupportedSubstitution.java:1]: This shouldn't be reverted.

**Review by gsmet — CHANGES_REQUESTED** [review]
Added a small gripe.

File: extensions/tika/runtime/src/main/java/io/quarkus/tika/graalvm/CleanerNotSupportedSubstitution.java (hunk 1 above)
Comment (zakkak) [reply extensions/tika/runtime/src/main/java/io/quarkus/tika/graalvm/CleanerNotSupportedSubstitution.java]: Resolved in https://github.com/quarkusio/quarkus/pull/14457/commits/b09b7085a2622c3489b7b902f1dcd5805b5799e8

**Comment by gsmet** [comment]
Let's promote it then!

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Merged! Nice work @zakkak .

I think we can work on moving master to 21.0 now, will you do it?

**Comment by zakkak** [comment]
> Merged! Nice work @zakkak .

Thanks

> I think we can work on moving master to 21.0 now, will you do it?

Sure :)

**Pull request merged**